### PR TITLE
+act #17719 introduce Tuple3-22 classes for java api (master)

### PR DESCRIPTION
**Pull request by ktoso**
Refs https://github.com/akka/akka/issues/17719
Forward port of https://github.com/akka/akka/pull/17721

### CODE REVIEW

**Comment by 2m** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Pushed fix, will merge after validation

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
MiMa failure was:

```
[info] akka-actor: found 1 potential binary incompatibilities (filtered 232)
[error]  * method toString()java.lang.String in object akka.japi.Pair does not have a correspondent in new version
[error]    filter with: ProblemFilters.exclude[MissingMethodProblem]("akka.japi.Pair.toString")
```

Which does not seem to make sense, toString always was available on it, but now thanks to moving to case class it's a generated one.

File: project/MiMa.scala [hunk 1]
```diff
@@ -534,7 +534,10 @@ object MiMa extends AutoPlugin {
       FilterAnyProblem("akka.remote.RemoteWatcher$UnwatchRemote"),
       FilterAnyProblem("akka.remote.RemoteWatcher$Rewatch"),
       FilterAnyProblem("akka.remote.RemoteWatcher$RewatchRemote"),
-      FilterAnyProblem("akka.remote.RemoteWatcher$Stats")
+      FilterAnyProblem("akka.remote.RemoteWatcher$Stats"),
+
+      // toString is available on any object, mima is confused due to a generated toString appearing #17722
+      ProblemFilters.exclude[MissingMethodProblem]("akka.japi.Pair.toString")
```
Comment (rkuhn) [inline project/MiMa.scala]: Hmm, I don’t get it; what is the actual error message it prints?

Comment (ktoso) [inline project/MiMa.scala]: @rkuhn It's

```
[info] akka-actor: found 1 potential binary incompatibilities (filtered 232)
[error]  * method toString()java.lang.String in object akka.japi.Pair does not have a correspondent in new version
[error]    filter with: ProblemFilters.exclude[MissingMethodProblem]("akka.japi.Pair.toString")
```

https://github.com/akka/akka/pull/17722#issuecomment-114485665

Comment (rkuhn) [inline project/MiMa.scala]: I think this is a bug in MiMa, please report over there and move along, nothing to see ;-)

**Comment by rkuhn** [comment]
LGTM

File: project/MiMa.scala (hunk 1 above)
Comment (ktoso) [inline project/MiMa.scala]: Yup, opened a ticket already. Merging this in a sec

File: akka-actor/src/main/boilerplate/akka/japi/tuple/Tuples.scala.template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+[3..22#/**
+ * Used to create tuples with 1 elements in Java.
+ */
+object Tuple1 {
+  def create[[#T1#]]([#t1: T1#]) = new Tuple1([#t1#])
+}
+
+/**
+ * Java API Tuple container.
```
Comment (hepin1989) [inline akka-actor/src/main/boilerplate/akka/japi/tuple/Tuples.scala.template]: Tuple -> Tuple1

Comment (ktoso) [inline akka-actor/src/main/boilerplate/akka/japi/tuple/Tuples.scala.template]: Here it is in the sense of "it is a tuple", so without number

**Pull request merged**